cuda/solver/gmres_kernels.cu

```diff
@@ -738,12 +723,12 @@ void step_2(std::shared_ptr<const CudaExecutor> exec,
             const matrix::Dense<ValueType> *krylov_bases,
             const matrix::Dense<ValueType> *hessenberg,
             matrix::Dense<ValueType> *y, matrix::Dense<ValueType> *x,
```

Comment:
I think you don't need the `x` here either, so you can probably remove it from the signature of `step_2`.